source/postprocess/current_surface.cc

```diff
@@ -0,0 +1,346 @@
+                          // This is needed later to keep the vertices together when sorting.
+                          std::vector<double> vertex_row;
+                          for (unsigned int i=0; i<dim; ++i)
+                            vertex_row.push_back(vertex[i]);
+
+                          local_surface_height.push_back(vertex_row);
+                        }
+                    }
+
+          // Combine all local_surfaces and broadcast back.
+          std::vector<std::vector<double>> temp_surface =
+            Utilities::MPI::compute_set_union(local_surface_height, this->get_mpi_communicator());
+
+          // Sort the vector so that it ascends in X.
+          std::sort(temp_surface.begin(), temp_surface.end(), [](const std::vector<double> &a, const std::vector<double> &b)
+          {
+            return a[0] < b[0];
+          });
+
+          // Define a comparison to remove duplicate surface points.
+          const auto compareRows = [](const std::vector<double> &row1, const std::vector<double> &row2)
+          {
+            return row1 == row2;
+          };
+
+          // Remove non-unique rows from the sorted 2D vector
+          const auto last = std::unique(temp_surface.begin(), temp_surface.end(), compareRows);
+          temp_surface.erase(last, temp_surface.end());
+
+          // Resize data table.
+          TableIndices<dim-1> size_idx;
+          for (unsigned int d=0; d<dim-1; ++d)
+            size_idx[d] = temp_surface.size();
+
+          data_table.TableBase<dim-1,double>::reinit(size_idx);
+          TableIndices<dim-1> idx;
+
+          // Fill the data table with the y-values that correspond to the surface.
+          if (dim==2)
+            {
+              for (unsigned int x=0; x<(data_table.size()[0]); ++x)
+                {
+                  idx[0] = x;
+                  data_table(idx) = temp_surface[x][1];
+                }
+            }
+
+          if constexpr (dim == 2)
+            for (unsigned int x=0; x<(data_table.size()[0]); ++x)
+              data_table(x) = temp_surface[x][1];
+
+          // Fill the coordinates with the x-values used for the data table.
+          //std::array<std::vector<double>, dim-1> coordinates;
+          coordinates[0].clear();
+          for (unsigned int i=0; i<temp_surface.size(); ++i)
+            coordinates[0].push_back(temp_surface[i][0]);
+
+          // Serialize coordinates and data table in save function. In load function recreate surface function.
+          surface_function = std::make_unique<Functions::InterpolatedTensorProductGridData<dim-1>>(coordinates, data_table);
+        }
+
+      set_last_output_time (this->get_time());
+
+      // Can I override this to a void function so we don't need to print a pair?
+      const std::string filename_prefix ("update surface");
+      return std::make_pair (std::string ("Postprocess: "),
+                             filename_prefix);
```

Comment:
no this is the expected format for every postprocessor. But you can simplify it to the following:
```suggestion
      return std::make_pair ("Storing deformed surface topography: ",
                             "Done");
```